tasks/appraisal.rake

```diff
@@ -0,0 +1,187 @@
+      '3.0' => '3.0.4',
+      '3.1' => '3.1.2',
+      '3.2' => '3.2.0',
+      'jruby-9.2.8.0' => 'jruby-9.2.8.0',
+      'jruby-9.2' => 'jruby-9.2.21.0',
+      'jruby-9.3' => 'jruby-9.3.9.0',
+      'jruby-9.4' => 'jruby-9.4.0.0',
+    }[ruby_version]
+
+    return "ruby_#{ruby_version}" if ruby_version =~ /^\d/
+
+    ruby_version.tr('-', '_')
+  end
+
+  desc 'Generate Appraisal gemfiles. Takes a version list as argument, defaults to all'
+  task :generate do |_task, args|
+    ruby_versions(args.to_a).each do |ruby_version|
+      cmd = []
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+      cmd << ['gem', 'install', 'bundler', '-v', bundler_version(ruby_version)] if bundler_version(ruby_version)
+      cmd << [*bundle(ruby_version), 'config', 'without', 'check']
+      cmd << [*bundle(ruby_version), 'install']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'generate']
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+
+      cmd = cmd.map { |c| c << '&&' }.flatten.tap(&:pop)
+
+      p cmd
+      p docker(ruby_version, cmd)
+      sh docker(ruby_version, cmd).join(' ')
+    end
+  end
+
+  desc 'Install Appraisal gemfiles. Takes a version list as argument, defaults to all'
+  task :install do |_task, args|
+    ENV['APPRAISAL_SKIP_BUNDLE_CHECK'] = 'y'
+
+    ruby_versions(args.to_a).each do |ruby_version|
+      cmd = []
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+      cmd << ['gem', 'install', 'bundler', '-v', bundler_version(ruby_version)] if bundler_version(ruby_version)
+      cmd << [*bundle(ruby_version), 'config', 'without', 'check']
+      cmd << [*bundle(ruby_version), 'install']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'generate']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'install']
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+
+      cmd = cmd.map { |c| c << '&&' }.flatten.tap(&:pop)
+
+      p cmd
+      p docker(ruby_version, cmd)
+      sh docker(ruby_version, cmd).join(' ')
+    end
+
+    ENV.delete('APPRAISAL_SKIP_BUNDLE_CHECK')
+  end
+
+  desc 'Update Appraisal gemfiles. Takes a version list as argument, defaults to all'
+  task :update do |_task, args|
+    ruby_versions(args.to_a).each do |ruby_version|
+      cmd = []
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+      cmd << ['gem', 'install', 'bundler', '-v', bundler_version(ruby_version)] if bundler_version(ruby_version)
+      cmd << [*bundle(ruby_version), 'config', 'without', 'check']
+      cmd << [*bundle(ruby_version), 'install']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'generate']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'update']
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+
+      cmd = cmd.map { |c| c << '&&' }.flatten.tap(&:pop)
+
+      p cmd
+      p docker(ruby_version, cmd)
+      sh docker(ruby_version, cmd).join(' ')
+    end
+  end
+
+  desc 'Add all platforms to Appraisal gemfiles. Takes a version list as argument, defaults to all'
+  task :platform do |_task, args|
```

Comment:
Should this be inlined on every task that generates a gemfile? E.g. so that we never forget to run it?

Reply:
No, because once the `PLATFORMS` section is set in a lockfile, it does not need to be re-run: bundler will resolve gems for all platforms listed in that section. So it's basically a one-shot to be run when we regenerate everything from scratch, but I would rather have this written out as an actual task rather than run these commands by hand.

Reply:
OK, seems reasonable :+1: